flocker/node/agents/test/test_ebs.py

```diff
@@ -189,6 +192,23 @@ def test_unexpected_device_discovered(self, device):
             exception,
         )
 
+    @capture_logging(assertHasMessage, NO_NEW_DEVICE_IN_OS)
+    def test_no_new_device_logged(self, logger):
+        """
+        ``NO_NEW_DEVICE_IN_OS`` is logged if a new device does not appear.
+        """
+        self.assertRaises(
+            AttachedUnexpectedDevice,
+            _attach_volume_and_wait_for_device,
+            volume=self.volume,
+            attach_to=self.compute_id,
+            attach_volume=lambda *a, **kw: None,
+            detach_volume=lambda *a, **kw: None,
+            device=b'/dev/sdh',
+            blockdevices=[],
+            time_limit=0,
+        )
+
```

Comment:
Yep, testing `_wait_for_device` directly is much simpler and doesn't require any changes to the time_limit arguments. I've moved the test to to a new `WaitForDeviceTests` case.